integration_test_suite/integration_tests/src/main/java/io/prometheus/jmx/test/httpserver/general/BlackListAndWhiteListObjectNames_IT.java

```diff
@@ -0,0 +1,210 @@
+
+        // Get the id to force the network creation
+        network.getId();
+    }
+
+    @TestEngine.BeforeAll
+    public void beforeAll() throws Exception {
+        // Application container
+        applicationContainer = new GenericContainer<>(dockerImageName)
+                .waitingFor(Wait.forLogMessage(".*Running.*", 1))
+                .withClasspathResourceMapping("common", "/temp", BindMode.READ_ONLY)
+                .withClasspathResourceMapping(getClass().getName().replace(".", "/"), "/temp", BindMode.READ_ONLY)
+                .withCommand("/bin/sh application.sh")
+                .withExposedPorts(9999)
+                .withLogConsumer(outputFrame -> System.out.print(outputFrame.getUtf8String()))
+                .withNetwork(network)
+                .withNetworkAliases("application")
+                .withStartupTimeout(Duration.ofMillis(30000))
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .withWorkingDirectory("/temp");
+
+        if (DockerImageNameParameters.isJava6(dockerImageName)) {
+            applicationContainer.withCommand("/bin/sh application_java6.sh");
+        }
+
+        applicationContainer.start();
+
+        // Exporter container
+        exporterContainer = new GenericContainer<>(dockerImageName)
+                .waitingFor(Wait.forLogMessage(".*Running.*", 1))
+                .withClasspathResourceMapping("common", "/temp", BindMode.READ_ONLY)
+                .withClasspathResourceMapping(getClass().getName().replace(".", "/"), "/temp", BindMode.READ_ONLY)
+                .withCommand("/bin/sh exporter.sh")
+                .withExposedPorts(8888)
+                .withLogConsumer(outputFrame -> System.out.print(outputFrame.getUtf8String()))
+                .withNetwork(network)
+                .withNetworkAliases("exporter")
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .withWorkingDirectory("/temp");
+
+        if (DockerImageNameParameters.isJava6(dockerImageName)) {
+            exporterContainer.withCommand("/bin/sh exporter_java6.sh");
+        }
+
+        exporterContainer.start();
+
+        // HTTP client
+        httpClient = new HttpClient("http://localhost:" + exporterContainer.getMappedPort(8888));
+    }
+
+    @TestEngine.Test
+    public void testHealthy() throws Exception {
+        String path = "/-/healthy";
+        Request.Builder requestBuilder = httpClient.createRequest(path);
+        try (Response response = httpClient.execute(requestBuilder)) {
+            assertThat(response).isNotNull();
+            assertThat(response.code()).isEqualTo(200);
+            ResponseBody responseBody = response.body();
+            assertThat(responseBody).isNotNull();
+            String content = responseBody.string();
+            assertThat(content).isNotNull();
+            assertThat(content).isEqualTo("Exporter is Healthy.");
+        }
+    }
+
+    @TestEngine.Test
```

Comment:
Using this random code line to discuss antublue vs JUnit: Glad to hear that antublue is also available on Maven central, not depending on the antublue repository will make the build a lot more stable.

I don't want to be too picky with test dependencies. With runtime dependencies we should be very restrictive, as they might cause trouble in instrumented applications. Test dependencies are not critical.

However, I found one more disadvantage of antublue compared to JUnit: I'm using Intellij (as many Java developers), and Intellij doesn't seem to recognize antublue tests. If I click the little green play button for a test it just says "No tests were found". Being able to run a test directly in Intellij is a strong argument for JUnit. What do you think?

Reply:
One issue with using JUnit 4 is there are no annotations for something to be run before all tests / after all tests that are non-static.

`@BeforeClass` is/needs to be static to set up the Docker network once for performance reasons

`@Before` is run before every test.

Assuming the tests methods are run on the same thread, a `ThreadLocal` in the `@BeforeClass` could be used to set up state for `@Test` methods

---

The latest version (v2.0.6) of the AntuBLUE Test Engine supports execution via IntelliJ. (Should be on Maven Central within a few hours.)

Reply:
This has been primarily addressed in AntuBLUE Test Engine v3.0.0, which a minor issue that appears to be an IntelliJ bug.

https://youtrack.jetbrains.com/issue/IDEA-317561/IntelliJ-test-method-selections-fails-for-hierarchical-test-in-test-output-window